lib/cisco_node_utils/fabricpath_global.rb

```diff
@@ -44,12 +49,27 @@ def self.fabricpath_feature
       fail 'fabricpath_feature not found' if fabricpath.nil?
       return :disabled if fabricpath.nil?
       fabricpath.to_sym
+    rescue Cisco::CliError => e
+      # cmd will syntax reject when feature is not enabled
+      raise unless e.clierror =~ /Syntax error/
+      return :disabled
     end
 
     def self.fabricpath_feature_set(fabricpath_set)
       curr = FabricpathGlobal.fabricpath_feature
       return if curr == fabricpath_set
 
+      if Vdc.vdc_support
+        # For modular platforms, make sure to limit the feature to
+        # modules which support this feature
+        if fabricpath_set == :installed || fabricpath_set == :enabled
+          v = Vdc.new('default')
+          v.limit_resource_module_type =
+            config_get('fabricpath', 'supported_modules')
```

Comment:
I like the idea of specifying module type explicitly through the vdc provider. We just want to avoid the older line-cards in the vdc. The limit-resource method (limit_resource_module_type=) already does the interface allocation. I will remove this.